libraries/chain/worker_proposal_evaluators.cpp

```diff
@@ -0,0 +1,63 @@
+#include <golos/protocol/worker_proposal_operations.hpp>
+#include <golos/protocol/exceptions.hpp>
+#include <golos/chain/steem_evaluator.hpp>
+#include <golos/chain/database.hpp>
+#include <golos/chain/steem_objects.hpp>
+#include <golos/chain/worker_proposal_objects.hpp>
+
+namespace golos { namespace chain {
+
+    void worker_proposal_create_evaluator::do_apply(const worker_proposal_create_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_20__1013, "worker_proposal_create_operation");
+
+        const auto& comment = _db.get_comment(o.author, o.permlink);
+
+        GOLOS_CHECK_LOGIC(comment.parent_author == STEEMIT_ROOT_POST_PARENT,
+            logic_exception::worker_proposal_can_be_created_only_on_post,
+            "Worker proposal can be created only on post");
+
+        const auto now = _db.head_block_time();
+
+        auto after_post_sec = (now - comment.created).to_seconds();
+        GOLOS_CHECK_LOGIC(after_post_sec >= GOLOS_WORKER_PROPOSAL_AFTER_POST_SEC,
+            logic_exception::too_early_creating_worker_proposal,
+            "Too early creating worker proposal for this post"); 
```

Comment:
Not fully understand for what this limit.
I can't create proposal and post in one transaction?

Reply:
This restriction should check opposite condition: proposal can only be attached to post during 1st hour or so. The motivation for it is to disable attaching worker proposal to post when post is closing soon, and it's too late to vote (approve). It's easier synchronise proposal approvement time with post cashout window, so closing post will also close proposal, which got not enough approvements.
Actually, it's some kind of "fool protection", so we can remove it